content/assignments/minishell/checker/_test/run_test.sh

```diff
@@ -0,0 +1,244 @@
+#!/bin/bash
+# SPDX-License-Identifier: BSD-3-Clause
+
+# ----------------- General declarations and util functions ------------------ #
+
+exec_name="mini-shell"
+ref_name="bash"
+ref_name_alt="dash"
+
+MAIN_TEST_DIR="_test/outputs"
+INPUT_DIR="_test/inputs"
+REFS_DIR="_test/refs"
+LOG_FILE="/dev/null"
+export max_points=100
+TEST_TIMEOUT=30
+
+TEST_LIB=_test/test_lib.sh
+export CHECK_SKIP='parser.\(\(tab\|yy\).\(c\|h\)\|h\)$'
+
+# Load the lib functions.
+if ! [ -e "$TEST_LIB" ]; then
+	echo "Test library not found. Check \$TEST_LIB ($TEST_LIB)"
+	exit 1
+fi
+source "$TEST_LIB"
+
+export SHELL_PROMPT="> "
+
+# Change this of you want to keep the logs after execution.
+DO_CLEANUP=no
+
+BUFFERING_WRAPPER=""
+if command -v stdbuf &> /dev/null; then
+	# stdbuf may not be installed on cygwin.
+	BUFFERING_WRAPPER="stdbuf -i 0"
+fi
+
+USE_VALGRIND=yes
+VALGRIND_LOG="valgrind.log"
+
+# ----------------- Init and cleanup tests ----------------------------------- #
+
+# Initializes a test.
+init_test()
+{
+
+	if ! [ -e "$exec_name" ]; then
+		echo "$exec_name not found! Cannot run the test"
+		exit 1
+	fi
+
+	if [ "$USE_VALGRIND" = "yes" ]; then
+		which valgrind &> /dev/null
+		if [ $? -eq 0 ]; then
```

Comment:
Simply use the result of the command, instead of checking the return code separately.
```suggestion
		if which valgrind &> /dev/null; then
```

Also, for the `stdbuf` check above, the `command` builtin is used. It may be worth using only one of them across the entire script.